### ISSUE TITLE
False positive for E1120 (no-value-for-parameter)

### ISSUE THREAD

**Issue by bje-** [PyCQA/pylint#3632]
### Steps to reproduce
1. Run pylint over the following small  test case:

```
"""docstring"""

class PV1Axis():
    """Single-axis tracking PV."""
    def __init__(self, polygon, capacity, filename, column, build_limit=None, label='PV'):
        pass

class Biofuel():
    """Model of open cycle gas turbines burning biofuel."""
    def __init__(self, polygon, capacity, label='biofuel'):
        pass

for g in [PV1Axis, Biofuel]: # nb: works if I reverse the list order
    if g == Biofuel:
        g(0, 100)
```

### Current behavior

```
example.py:15: [E1120(no-value-for-parameter), ] No value for argument 'filename' in constructor call
example.py:15: [E1120(no-value-for-parameter), ] No value for argument 'column' in constructor call
```

### Expected behavior

No warnings should be generated.

### pylint --version output

```
pylint 2.5.2
astroid 2.4.1
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0]
```

**Comment by bje-** [comment]
Updated with a small, standalone test case.

**Comment by bje-** [comment]
Still present in Pylint 2.8.3.

**Comment by bje-** [comment]
This was fixed somewhere along the way. Works in Pylint 2.14.5.

**Comment by Pierre-Sassoulas** [comment]
Thank you for checking @bje- , I must have made a mistake when labeling on july 2, or we regressed since 2.14.5. I'm reopening so I check what the problem was before releasing 2.15.0.

**Comment by Pierre-Sassoulas** [comment]
Also still an issue on current main.
